components/eam/src/chemistry/aerosol/wetdep.F90

```diff
@@ -502,798 +471,732 @@ subroutine wetdepa_v2( ncol, deltat, &
+         src1 = 0._r8
+      endif
+      src2 = sol_factb *cldv_ik*odds*tracer_2/deltat
 
-      end do main_k_loop ! End of k = 1, pver
+   endif
 
-   end subroutine wetdepa_v2
+   src = src1 + src2             ! total stratiform or convective scavenging
+   fin=src1/(src + small_value)    ! fraction taken by incloud processes
+
+   end subroutine wetdep_scavenging
 
+!==============================================================================
+   subroutine wetdep_resusp(                         &
+                        is_st_cu,       mam_prevap_resusp_optcc,& ! in
+                        pdel_ik,        evapx,                  & ! in
+                        precabx_old,    precabx_base_old,       & ! in
+                        scavabx_old,    precnumx_base_old,      & ! in
+                        precabx_new,    precabx_base_new,       & ! out
+                        scavabx_new,    precnumx_base_new,      & ! out
+                        resusp_x                                ) ! out
+
+! ------------------------------------------------------------------------------
+! do precip production, resuspension and scavenging
+! ------------------------------------------------------------------------------
+   integer, intent(in) :: is_st_cu ! options for stratiform (1) or convective (2) clouds
+                                        ! raindrop size distribution is
+                                        ! different for different cloud:
+                                        ! 1: assume marshall-palmer distribution
+                                        ! 2: assume log-normal distribution
+   integer, intent(in) :: mam_prevap_resusp_optcc       ! suspension options
+   real(r8),intent(in) :: pdel_ik       ! pressure thikness at current column and level [Pa]
+   real(r8),intent(in) :: evapx         ! evaporation at current layer [kg/kg/s]
+   real(r8),intent(in) :: precabx_base_old ! input of precipitation at cloud base [kg/m2/s]
+   real(r8),intent(in) :: precabx_old ! input of precipitation above this layer [kg/m2/s]
+   real(r8),intent(in) :: scavabx_old ! input of scavenged tracer flux from above [kg/m2/s]
+   real(r8),intent(in) :: precnumx_base_old ! input of precipitation number at cloud base [#/m2/s]
+   real(r8),intent(out) :: precabx_base_new ! output of precipitation at cloud base [kg/m2/s]
+   real(r8),intent(out) :: precabx_new ! output of precipitation above this layer [kg/m2/s]
+   real(r8),intent(out) :: scavabx_new ! output of scavenged tracer flux from above [kg/m2/s]
+   real(r8),intent(out) :: precnumx_base_new ! output of precipitation number at cloud base [#/m2/s]
+   real(r8),intent(out) :: resusp_x    ! aerosol mass re-suspension in a particular layer [kg/m2/s]
+
+
+   ! local variables
+   real(r8)            :: tmpa ! temporary working variable
+   real(r8), parameter :: prec_smallaa = 1.0e-30_r8  ! 1e-30 kg/m2/s (or mm/s) = 3.2e-23 mm/yr
```

Comment:
This parameter appears in a few different places within this module. Should it be a module parameter instead of living within multiple subroutines?

Reply:
I agree. We should make it a module parameter variable. If it mean differently in each calculation or have different units, we can make a not about it where it is declared.